gensim/downloader.py

```diff
@@ -0,0 +1,298 @@
+    """Function for calculating checksum of a downloaded model/corpus.
+
+    Args:
+        folder_dir(string): Path to the downloaded model.
+
+    Returns:
+        string: It returns the value for the checksum for folder_dir directory
+    """
+    hash_md5 = hashlib.md5()
+    for filename in os.listdir(folder_dir):
+        file_dir = os.path.join(folder_dir, filename)
+        with open(file_dir, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_md5.update(chunk)
+    return hash_md5.hexdigest()
+
+
+def info(dataset=None):
+    """Function for retrieving the list of corpora/models, if dataset is not provided. If dataset
+    is provided, then it gives detailed information about the dataset.
+
+    Args:
+        dataset(string): Name of the corpus/model.
+
+    Returns:
+        <class 'dict'>: It returns the models/corpora names with detailed information about each.
+    """
+    url = "https://raw.githubusercontent.com/chaitaliSaini/Corpus_and_models/master/list_with_filename.json"
+    response = urlopen(url)
+    data = response.read().decode("utf-8")
+    data = json.loads(data)
+    if dataset is not None:
+        corpora = data['gensim']['corpus']
+        models = data['gensim']['model']
+        if dataset in corpora:
+            logger.info("%s \n", data['gensim']['corpus'][dataset]["desc"])
+        elif dataset in models:
+            logger.info("%s \n", data['gensim']['model'][dataset]["desc"])
+        else:
+            raise Exception(
+                "Incorrect model/corpus name. Choose the model/corpus from the list "
+                "\n {}".format(json.dumps(data, indent=4)))
+    else:
+        return data
+
+
+def get_checksum(dataset):
+    """Function for retrieving the checksum of a corpus/model
+
+    Args:
+        dataset(string): Name of the corpus/model.
+
+    Returns:
+        string: It returns the checksum for corresponding the corpus/model.
+    """
+    data = info()
+    corpora = data['gensim']['corpus']
+    models = data['gensim']['model']
+    if dataset in corpora:
+        return data['gensim']['corpus'][dataset]["checksum"]
+    elif dataset in models:
+        return data['gensim']['model'][dataset]["checksum"]
+
+
+if not os.path.isfile(data_log_file_dir):
+    try:
+        logger.warning("Creating %s", data_log_file_dir)
+        data_log_file = open(data_log_file_dir, 'a')
+        initialize_data_log_file()
+    except:
+        raise Exception(
+            "Can't create {}. Make sure you have the read/write permissions "
+            "to the directory or you can try creating the file manually"
+            .format(data_log_file_dir))
+
+
+def _download(dataset):
+    """Function for downloading and installed dataset depending upon it's current status.
+
+    Args:
+        dataset(string): Name of the corpus/model.
+    """
+    url = "https://github.com/chaitaliSaini/Corpus_and_models/releases/download/{f}/{f}.tar.gz".format(f=dataset)
```

Comment:
Please add a FIXME comment so we don't merge such temporary constructs by accident.